File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    # 배열을 순회하면서 앞과 다른 수이면 새로운 배열에 추가
    new_arr = []
    previous = -1
    for num in arr:
        if num != previous:
            new_arr.append(num)
            previous = num
    #print(new_arr)
    return new_arr
        

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public static void log(Object o){
        System.out.println(o);
    }
    public int[] solution(int[] arr) {
        // 배열의 시작부터 순회하면서 스택에 push
        // 만약 peek한 값과 같으면 push하진 않음
        Stack<Integer> stack = new Stack();
        for(int num: arr){
            if(stack.isEmpty() || stack.peek() != num){
                stack.push(num);
            }
        }
        int[] result = stack.stream()
            .mapToInt(x -> x)
            .toArray();
        // for(int num: result){
        //     log(num);
        // }
        return result;
    }
}